cli/ops/tls.rs

```diff
@@ -42,6 +43,89 @@ struct ConnectTLSArgs {
   cert_file: Option<String>,
 }
 
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct StartTLSArgs {
+  rid: u32,
+  cert_file: Option<String>,
+  hostname: String,
+}
+
+pub fn op_start_tls(
+  state: &State,
+  args: Value,
+  _zero_copy: Option<ZeroCopyBuf>,
+) -> Result<JsonOp, OpError> {
+  let args: StartTLSArgs = serde_json::from_value(args)?;
+  let rid = args.rid as u32;
+  let cert_file = args.cert_file.clone();
+  let state_ = state.clone();
+
+  let mut domain = args.hostname;
+  if domain.is_empty() {
+    domain.push_str("localhost");
+  }
+
+  let op = async move {
+    let mut state = state_.borrow_mut();
+
+    let mut resource_holder =
+      match state.resource_table.take::<StreamResourceHolder>(rid) {
+        Some(resource) => *resource,
+        None => return Err(OpError::bad_resource_id()),
+      };
+
+    let tcp_stream = match resource_holder.resource {
+      StreamResource::TcpStream(ref mut stream) => stream.take(),
+      _ => return Err(OpError::bad_resource_id()),
+    };
```

Comment:
Why is it necessary to also call `stream.take()` here? Can't you just do
```rust
if let StreamResource::TcpStream(stream) = resource_holder.resource {
  // call the code below
} else {
  return Err(OpError::bad_resource_id())
}
```

Reply:
Because StreamResourceHolder implements Drop, it cannot be moved. I thought of putting it in Option so that TcpStream can be moved out, and tls_connector.connect requires ownership of tcpStream.